src/rust/bitbox02-rust/src/hww/api/set_device_name.rs

```diff
@@ -39,3 +39,60 @@ pub async fn process(
 
     Ok(Response::Success(pb::Success {}))
 }
+
+#[cfg(test)]
+mod tests {
+    extern crate std;
+    use super::*;
+
+    use crate::bb02_async::block_on;
+    use bitbox02::testing::{mock, Data};
+
+    #[test]
+    pub fn test_set_device_name() {
+        let some_name: std::string::String = "foo".into();
+
+        // All good.
+        mock(Data {
```

Comment:
@x1ddos if you could stomach a dependency, we could try https://docs.rs/mockall/0.8.0/mockall/ to make mocking more flexible and easy. for now the handrolled thing might work, we'll see

Reply:
doesn't look too bad so far, does it. but sure, if you get tired of mocking why not. it's only a dev dependency after all.

Reply:
as long as it doesn't pull in a large deps tree :)